src/compiler/checker.ts

```diff
@@ -13264,8 +13264,12 @@ namespace ts {
             return errorReported;
         }
 
-        function reportImplicitAnyError(noImplicitAny: boolean, declaration: Declaration, type: Type) {
+        function reportImplicitAny(noImplicitAny: boolean, declaration: Declaration, type: Type) {
             const typeAsString = typeToString(getWidenedType(type));
+            if (isInJSFile(declaration) && !getSourceFileOfNode(declaration).pragmas.has("checkJSDirective") && !compilerOptions.checkJs) {

```

Comment:
Do we even get here without checkJS? Also, this must be duplicate code of something?

Reply:
1. Yes, allowJS reports errors as usual, but doesn't actually display them.
2. Yes, `isCheckJsEnabledForFile`.

Reply:
So, why go out of our way to not report *this* error, if no errors will be shown anyway?

Reply:
Suggestions will still be shown, since they're stored in a separate array, and `errorOrSuggestion` at the bottom of the function will create a suggestion instead of an error in an allowJs file.